Posts/2021/03/Caps-lock-ToUpper.md

```diff
@@ -0,0 +1,206 @@
+---
+post_title: Can I Enable the Caps Lock Key?
+username: tfl@psp.co.uk
+Catagories: PowerShell
+tags: Caps Lock, string, ToUpper()
+Summary: How to enable all upper case input
+---
+
+**Q:** I have a script where users enter some information.
+This information needs to be entered in all capital letters, so my instructions say, “Please make sure the Caps Lock key is on before entering the information.”
+They don’t always do that, however.
+Is there a way to turn the Caps Lock key on and off using a script?
+
+**A:**  I don't know how to run the key off and on, but with PowerShell, there is a way to mimic the effect of turning on the Caps Lock key.
+
+## User Input Considered Harmful
+
+Let's start with the observation that all user input is harmful.
+One of my earliest IT heroes, Edsger Dijkstra, published a seminal letter [Go To Statement Considered Harmful](https://homepages.cwi.nl/~storm/teaching/reader/Dijkstra68.pdf) in 1968 which began the structured programming revolution.
+And this is one reason, by the way, why PowerShell has no goto statement.
+The phrase "Considered Harmful" is also a well-known phrase that has a Wikipedia entry at [Considered Harmful](https://en.wikipedia.org/wiki/Considered_harmful#:~:text=Considered%20harmful%20was%20popularized%20among,the%20day%20and%20advocated%20structured).
```

Comment:
Remove locale from wikipedia links. It will redirect to the browser's language.

```suggestion
The phrase "Considered Harmful" is also a well-known phrase that has a Wikipedia entry at [Considered Harmful](https://wikipedia.org/wiki/Considered_harmful#:~:text=Considered%20harmful%20was%20popularized%20among,the%20day%20and%20advocated%20structured).
```